tests/Doctrine/Tests/DBAL/Functional/ResultCacheTest.php

```diff
@@ -238,25 +238,25 @@ public function testFetchingAllRowsSavesCache(callable $fetchAll): void
     public static function fetchAllProvider(): iterable
     {
         yield 'fetchAll' => [
-            static function (ResultCacheStatement $statement): void {
+            static function (ResultStatement $statement): void {
                 $statement->fetchAll();
             },
         ];
 
         yield 'fetchAllAssociative' => [
-            static function (ResultCacheStatement $statement): void {
+            static function (ForwardCompatibility\Result $statement): void {
                 $statement->fetchAllAssociative();
             },
         ];
 
         yield 'fetchAllNumeric' => [
-            static function (ResultCacheStatement $statement): void {
+            static function (ForwardCompatibility\Result $statement): void {
                 $statement->fetchAllNumeric();
             },
         ];
 
         yield 'fetchFirstColumn' => [
-            static function (ResultCacheStatement $result): void {
+            static function (ForwardCompatibility\Result $result): void {
```

Comment:
https://github.com/doctrine/dbal/pull/4529#discussion_r593785306 applies to all the above cases.

Reply:
Not sure to understand here. ResultStament interface does not contain the fetchAllAssociative(), fetchAllNumeric() &  fetchFirstColumn() methods. I can use the Doctrine\DBAL\Result interface, if we prefer not relying on concrete object.

Reply:
Alright, so the expected type was not unnecessarily specific: it was the type that implemented both of the old and new APIs. It still implements them but now the `fetch*()` methods return another class that implements both interfaces as well. Looks good 👍